pkgs/tools/networking/strongswan/default.nix

```
@@ -41,6 +48,7 @@ stdenv.mkDerivation rec {
     ./ext_auth-path.patch
     ./firewall_defaults.patch
     ./updown-path.patch
+    cve_2023_26463_patch
```

```suggestion
    (fetchpatch {
      # NB: this patch is included in version 5.9.10
      name = "CVE-2023-26463.patch";
      url = "https://download.strongswan.org/security/CVE-2023-26463/strongswan-5.9.8-5.9.9_tls_auth_bypass_exp_pointer.patch";
      hash = "sha256-FrD7fXS8jRkdA99qhDhHo70AiBoAhuTJnw2a4WugeJg=";
    })
```

You can put `fetchpatch` directly into `patches` by enclosing it in round brackets. Also, please try to use hash wherever possible.

---

Okay, got it! Regarding `hash` - am I right that this is the general recommendation for all of nixpkgs?

---

Yes, most fetchers support `hash` these days. Few, like `fetchhg` still don't. No reason to hold us back elsewhere.